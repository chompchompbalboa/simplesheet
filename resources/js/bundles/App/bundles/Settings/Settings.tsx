//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import styled from 'styled-components'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const Settings = () => <Container>Settings</Container>

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
  position: fixed;
  top: 0;
  right: 0;
`

export default Settings
