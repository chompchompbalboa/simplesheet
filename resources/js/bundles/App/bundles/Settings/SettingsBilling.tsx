//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'

import ContentContent from '@app/bundles/Content/ContentContent'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const SettingsBilling = () => {

  return (
    <ContentContent>
      SettingsBilling
    </ContentContent>
  )
}

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default SettingsBilling
