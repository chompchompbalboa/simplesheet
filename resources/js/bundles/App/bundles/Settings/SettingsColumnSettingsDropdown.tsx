//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import styled from 'styled-components'

import { ISheetColumnTypeDropdownOption } from '@app/state/sheet/types'

//-----------------------------------------------------------------------------
// Components
//-----------------------------------------------------------------------------
const SheetSettingsContentColumnTypesDropdown = ({
  data: {
    options
  }
}: SheetSettingsContentColumnTypesDropdownProps) => {

  return (
    <Container>
      {Object.keys(options) && Object.keys(options).map(optionId => {
        const option = options[optionId]
        return (
          <DropdownOption
            key={option.id}>
            {option.value}
          </DropdownOption>
        )
      })}
    </Container>
  )
}

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
interface SheetSettingsContentColumnTypesDropdownProps {
  data: {
    options: { [id: string]: ISheetColumnTypeDropdownOption }
  }
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
`

const DropdownOption = styled.div``

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default SheetSettingsContentColumnTypesDropdown
