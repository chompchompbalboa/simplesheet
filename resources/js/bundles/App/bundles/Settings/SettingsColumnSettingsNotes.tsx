//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import styled from 'styled-components'

//-----------------------------------------------------------------------------
// Components
//-----------------------------------------------------------------------------
const SheetSettingsContentColumnTypesNotes = ({
}: SheetSettingsContentColumnTypesNotesProps) => (
  <Container>
    Notes
  </Container>
)

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
interface SheetSettingsContentColumnTypesNotesProps {
  data: any
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
`

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default SheetSettingsContentColumnTypesNotes
