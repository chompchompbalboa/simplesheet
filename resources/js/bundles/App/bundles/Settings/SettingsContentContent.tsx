//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import styled from 'styled-components'

//-----------------------------------------------------------------------------
// Components
//-----------------------------------------------------------------------------
const SheetSettingsContentContent = ({
  children,
  testId = 'SheetSettingsContentContent'
}: SheetSettingsContentContentProps) => (
  <Container
    data-testid={testId}>
    {children}
  </Container>
)

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
interface SheetSettingsContentContentProps {
  children?: any
  testId?: string
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
  height: 100%;
  width: 100%;
`

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default SheetSettingsContentContent
