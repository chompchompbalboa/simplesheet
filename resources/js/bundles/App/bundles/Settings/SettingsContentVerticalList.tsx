//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import styled from 'styled-components'

//-----------------------------------------------------------------------------
// Components
//-----------------------------------------------------------------------------
const SheetSettingsContentColumn = ({
  children
}: SheetSettingsContentColumnProps) => (
  <Container>
    {children}
  </Container>
)

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
interface SheetSettingsContentColumnProps {
  children?: any
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
  height: 100%;
  width: 13rem;
  border-right: 1px solid rgb(220, 220, 220);
  display: flex;
  flex-direction: column;
  overflow: hidden;
`

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default SheetSettingsContentColumn
