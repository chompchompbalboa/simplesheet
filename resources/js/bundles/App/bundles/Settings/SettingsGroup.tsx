//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React, { useState } from 'react'
import { useSelector } from 'react-redux'
import styled from 'styled-components'

import { ARROW_DOWN, ARROW_RIGHT }  from '@app/assets/icons'

import { IAppState } from '@app/state'

import Icon from '@/components/Icon'

//-----------------------------------------------------------------------------
// Component
//-----------------------------------------------------------------------------
const SettingsGroup = ({
  children,
  header
}: ISettingsGroup) => {
  
  const [ isContentVisible, setIsContentVisible ] = useState(true)
  const userColorPrimary = useSelector((state: IAppState) => state.user.color.primary)

  return (
    <Container>
      <Header
        isContentVisible={isContentVisible}
        onClick={() => setIsContentVisible(!isContentVisible)}
        userColorPrimary={'rgb(225, 225, 225)' || userColorPrimary}>
        <Icon
          icon={isContentVisible ? ARROW_DOWN : ARROW_RIGHT}/>
        &nbsp;&nbsp;{header}
      </Header>
      <Content
        isContentVisible={isContentVisible}>
        {children}
      </Content>
    </Container>
  )
}

export interface ISettingsGroup {
  children?: any
  header: string
}

//-----------------------------------------------------------------------------
// Styled Components
//-----------------------------------------------------------------------------
const Container = styled.div`
  width: 100%;
  background-color: rgb(245, 245, 245);
  border: 1px solid rgb(220, 220, 220);
  border-radius: 4px;
  margin-bottom: 0.5rem;
`

const Header = styled.div`
  cursor: pointer;
  padding: 0.5rem 0.25rem;
  display: flex;
  align-items: center;
  background-color: ${ ({ userColorPrimary }: IHeader ) => userColorPrimary };
  color: black;
  font-size: 0.9rem;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  border-bottom-left-radius: ${ ({ isContentVisible }: IHeader ) => isContentVisible ? '0' : '4px' };
  border-bottom-right-radius: ${ ({ isContentVisible }: IHeader ) => isContentVisible ? '0' : '4px' };
`
interface IHeader {
  isContentVisible: boolean
  userColorPrimary: string
}

const Content = styled.div`
  padding: 0.5rem;
  display: ${ ({ isContentVisible }: IContent ) => isContentVisible ? 'block' : 'none' };
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
`
interface IContent {
  isContentVisible: boolean
}

export default SettingsGroup
