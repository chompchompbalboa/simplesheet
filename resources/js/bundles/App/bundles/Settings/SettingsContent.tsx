//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'

import { ISettingsContent } from '@app/bundles/Settings/Settings'

import SettingsBilling from '@app/bundles/Settings/SettingsBilling'
import SettingsUser from '@app/bundles/Settings/SettingsUser'
import SettingsTeams from '@app/bundles/Settings/SettingsTeams'

//-----------------------------------------------------------------------------
// Components
//-----------------------------------------------------------------------------
const SettingsContent = ({
  activeContent
}: SettingsContentProps) => {

  const content = {
    BILLING: SettingsBilling,
    TEAMS: SettingsTeams,
    USER: SettingsUser,
  }

  const ContentComponent = activeContent ? content[activeContent] : null

  return <ContentComponent />
}

//-----------------------------------------------------------------------------
// Props
//-----------------------------------------------------------------------------
interface SettingsContentProps {
  activeContent: ISettingsContent
}

//-----------------------------------------------------------------------------
// Export
//-----------------------------------------------------------------------------
export default SettingsContent
