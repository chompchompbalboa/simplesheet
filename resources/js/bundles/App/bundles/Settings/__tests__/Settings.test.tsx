//-----------------------------------------------------------------------------
// Imports
//-----------------------------------------------------------------------------
import React from 'react'
import 'jest-styled-components'
import { cleanup, fireEvent, render, renderWithRedux } from '@app/utils/tests'
import Settings from '../Settings'

//-----------------------------------------------------------------------------
// Setup
//-----------------------------------------------------------------------------
afterEach(cleanup)

//-----------------------------------------------------------------------------
// Tests
//-----------------------------------------------------------------------------
test("Renders correctly", async () => {
  const { container } = render(<Settings />)
  expect(container).toMatchSnapshot()
})

test("Opens and closes when clicking the link", async () => {
  const { getByTestId } = render(<Settings />)
  const settingsLink = getByTestId('settingsLink')
  const settingsContainer = getByTestId('settingsContainer')
  expect(settingsContainer).toHaveStyleRule('left', '100vw')
  fireEvent.click(settingsLink)
  expect(settingsContainer).toHaveStyleRule('left', '75vw')
  fireEvent.click(settingsLink)
  expect(settingsContainer).toHaveStyleRule('left', '100vw')
})

test("Transforms the link upon opening and closing", async () => {
  const { getByTestId } = renderWithRedux(<Settings />)
  const settingsLink = getByTestId('settingsLink')
  expect(settingsLink).toHaveStyleRule('transform', 'scale(-1,1)')
  fireEvent.click(settingsLink)
  expect(settingsLink).toHaveStyleRule('transform', 'none')
  fireEvent.click(settingsLink)
  expect(settingsLink).toHaveStyleRule('transform', 'scale(-1,1)')
})