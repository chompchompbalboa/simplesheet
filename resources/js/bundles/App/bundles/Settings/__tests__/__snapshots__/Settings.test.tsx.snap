// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
.c3 {
  cursor: pointer;
  width: 100%;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c4 {
  display: none;
}

.c0 {
  z-index: 3;
  position: fixed;
  top: 0;
  right: 0;
  cursor: pointer;
  padding: 0.375rem;
  color: rgb(80,80,80);
  -webkit-transform: scale(-1,1);
  -ms-transform: scale(-1,1);
  transform: scale(-1,1);
  -webkit-transition: -webkit-transform 0.25s;
  -webkit-transition: transform 0.25s;
  transition: transform 0.25s;
}

.c0:hover {
  color: rgb(40,40,40);
}

.c1 {
  z-index: 2;
  position: fixed;
  background-color: rgb(250,250,250);
  opacity: 0;
  top: 0;
  left: 100vw;
  height: 100vh;
  width: 25vw;
  padding: 1rem;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.c2 {
  margin-bottom: 1rem;
}

<div>
  <div
    class="c0"
    data-testid="settingsLink"
  >
    <svg
      height="1.5rem"
      viewBox="0 0 1024 1024"
      width="1.5rem"
    >
      <path
        d="M384 853.333h-170.667c-5.845 0-11.349-1.152-16.299-3.2-5.205-2.133-9.899-5.333-13.867-9.301s-7.125-8.661-9.301-13.867c-2.048-4.949-3.2-10.453-3.2-16.299v-597.333c0-5.845 1.152-11.349 3.2-16.299 2.133-5.205 5.333-9.899 9.301-13.867s8.661-7.125 13.867-9.301c4.949-2.048 10.453-3.2 16.299-3.2h170.667c23.552 0 42.667-19.115 42.667-42.667s-19.115-42.667-42.667-42.667h-170.667c-17.28 0-33.835 3.456-48.981 9.728-15.659 6.485-29.739 16-41.515 27.776s-21.291 25.856-27.776 41.515c-6.272 15.147-9.728 31.701-9.728 48.981v597.333c0 17.28 3.456 33.835 9.728 48.981 6.485 15.701 16 29.781 27.776 41.557s25.856 21.291 41.557 27.776c15.104 6.229 31.659 9.685 48.939 9.685h170.667c23.552 0 42.667-19.115 42.667-42.667s-19.115-42.667-42.667-42.667zM793.003 469.333h-409.003c-23.552 0-42.667 19.115-42.667 42.667s19.115 42.667 42.667 42.667h409.003l-140.501 140.501c-16.683 16.683-16.683 43.691 0 60.331s43.691 16.683 60.331 0l213.333-213.333c4.096-4.096 7.168-8.832 9.259-13.867 2.091-5.12 3.2-10.539 3.243-15.957 0.043-5.675-1.024-11.349-3.243-16.64-2.091-5.035-5.163-9.771-9.259-13.867l-213.333-213.333c-16.683-16.683-43.691-16.683-60.331 0s-16.683 43.691 0 60.331z"
        style="fill: currentColor;"
      />
    </svg>
  </div>
  <div
    class="c1"
    data-testid="settingsContainer"
  >
    <h2
      class="c2"
    >
      Settings
    </h2>
    <div
      class=""
    >
      <h3
        class="c3"
        data-testid="settingsTileHeader"
      >
        Colors
      </h3>
      <div
        class="c4"
        data-testid="settingsTileItems"
      >
        <div
          class=""
        >
          Primary
        </div>
        <div
          class=""
        >
          Secondary
        </div>
        <div
          class=""
        >
          Tertiary
        </div>
      </div>
    </div>
  </div>
</div>
`;
